### ISSUE TITLE
@storybook/semver lacks types that are available in upstream semver package

### ISSUE THREAD

**Issue by JReinhold** [storybookjs/storybook#19197]
## tl;dr
Our usage of [`@storybook/semver`](v) is causing minor annoyances with types. Instead of fixing the types we could attempt to replace the usage of our own fork with the actual upstream [`semver`](https://github.com/npm/node-semver) that has types via [`@types/semver`](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/semver).

## The problem

Around 2 years ago we forked `npm/node-semver` and uses it a few places as `@storybook/semver`. See #11008. The main purpose for this was to support IE11 and Yarn 2 better, because `semver` relied on `core-js`. Storybook v7 is dropping support for IE11, so we should do an attempt at using the upstream version instead which is up to date and has types.

## The proposed solution

1. Replace all dependencies on `@storybook/semver` with `semver` and `@types/semver`
2. Replace usages.
3. A full repo-wide search on `@storybook/semver` shouldn't yield any results.
4. Ensure it doesn't break anything. Especially building and using with Yarn 2+ should be tested.
5. Deprecate https://github.com/storybookjs/browser-semver once done.

**Comment by vorbrodt** [comment]
Hello, new contributor here! :) 

I would like to be assigned to this issue.

**Comment by JReinhold** [comment]
Welcome @vorbrodt, it's all yours!

Let me know if you need any help.

**Comment by danielamram** [comment]
Hey @JReinhold, i'v noticed no one is assigned to the issue and [uploaded a PR](https://github.com/storybookjs/storybook/pull/19292) can you have a look? (first contribution PR here)

Also regarding bullet 5: Deprecate [storybookjs/browser-semver](https://github.com/storybookjs/browser-semver) what should be done to deprecate it?

Thanks

**Comment by JReinhold** [comment]
Nice @danielamram, thanks for the PR!

> Also regarding bullet 5: Deprecate [storybookjs/browser-semver](https://github.com/storybookjs/browser-semver) what should be done to deprecate it?

We just need to mark the package as deprecated in the npm registry and do the same for the GitHub repository, or delete it. That's not anything you can do, I'll handle it once the PR has been merged.

**Comment by danielamram** [comment]
No worries, gussed it's a manual work with privileges to do so.
Anyways hit me up if I can help with anything, i'd be glad to help

**Comment by JReinhold** [comment]
@ndelangen can I get you to archive the GitHub repo [storybookjs/browser-semver](https://github.com/storybookjs/browser-semver)? I guess we don't want to deprecate the NPM package since 6.5 users will keep using it and it would be an annoying warning to display for them.

**Comment by shilman** [comment]
Egads!! I just released https://github.com/storybookjs/storybook/releases/tag/v7.0.0-alpha.37 containing PR #19292 that references this issue. Upgrade today to the `@next` NPM tag to try it out!

```
npx sb upgrade --prerelease
```

Closing this issue. Please re-open if you think there's still more to do.
